packages/x-charts/src/internals/plugins/featurePlugins/useChartCartesianAxis/useChartCartesianAxisPosition.selectors.ts

```diff
@@ -0,0 +1,112 @@
+import { createSelector } from '@mui/x-internals/store';
+import { type ComputedAxis } from '../../../../models/axis';
+import { selectorChartXAxis, selectorChartYAxis } from './useChartCartesianAxisRendering.selectors';
+import { selectorChartSeriesProcessed } from '../../corePlugins/useChartSeries';
+import { getBandSize } from '../../../../internals/getBandSize';
+import { isBandScale } from '../../../../internals/scaleGuards';
+import { getDataIndexForOrdinalScaleValue } from '../../../../internals/invertScale';
+import { type BarItemIdentifier, type SeriesId } from '../../../../models';
+
+export const selectorBarItemAtPosition = createSelector(
+  selectorChartXAxis,
+  selectorChartYAxis,
+  selectorChartSeriesProcessed,
+  function selectorBarItemAtPosition(
+    { axis: xAxes, axisIds: xAxisIds },
+    { axis: yAxes, axisIds: yAxisIds },
+    processedSeries,
+    svgPoint: Pick<DOMPoint, 'x' | 'y'>,
+  ): BarItemIdentifier | undefined {
+    const { series, stackingGroups = [] } = processedSeries?.bar ?? {};
+    const defaultXAxisId = xAxisIds[0];
+    const defaultYAxisId = yAxisIds[0];
+
+    let item: { dataIndex: number; seriesId: SeriesId } | undefined = undefined;
+
+    for (let stackIndex = 0; stackIndex < stackingGroups.length; stackIndex += 1) {
+      const group = stackingGroups[stackIndex];
+      const seriesIds = group.ids;
+
+      for (const seriesId of seriesIds) {
+        const aSeries = (series ?? {})[seriesId];
+
+        const xAxisId = aSeries.xAxisId ?? defaultXAxisId;
+        const yAxisId = aSeries.yAxisId ?? defaultYAxisId;
+
+        const xAxis = xAxes[xAxisId];
+        const yAxis = yAxes[yAxisId];
+
+        const bandAxis = aSeries.layout === 'horizontal' ? yAxis : xAxis;
+        const continuousAxis = aSeries.layout === 'horizontal' ? xAxis : yAxis;
+        const bandScale = bandAxis.scale;
+        const svgPointBandCoordinate = aSeries.layout === 'horizontal' ? svgPoint.y : svgPoint.x;
+
+        if (!isBandScale(bandScale)) {
+          continue;
+        }
+
+        const dataIndex = getDataIndexForOrdinalScaleValue(bandScale, svgPointBandCoordinate);
+
+        const { barWidth, offset } = getBandSize(
+          bandScale.bandwidth(),
+          stackingGroups.length,
+          (bandAxis as ComputedAxis<'band'>).barGapRatio,
+        );
+
+        const barOffset = stackIndex * (barWidth + offset);
+        const bandValue = bandAxis.data?.[dataIndex];
+
+        if (bandValue == null) {
+          continue;
+        }
+
+        const bandStart = bandScale(bandValue);
+
+        if (bandStart == null) {
+          continue;
+        }
+
+        const bandBarStart = bandStart + barOffset;
+        const bandBarEnd = bandBarStart + barWidth;
```

Comment:
Using Math.round() for bandBarStart and bandBarEnd will remove the highlight flickering.

I played with it end ended up with no highlight because the pointer was at 196px and band start at 196.003px

Reply:
We could, but I'm not sure it would solve anything. When zoomed in, users probably want a positive gap ratio, so they would end up changing it, which would make this break again. 

I think that ignoring the gap ratio below X pixels would be the way to go.